<launch>

    <include file="$(find quad_utils)/launch/load_params.launch" />

    <arg name="global_planner" default="twist" />
    <arg name="local_planner" default="full" />
    <arg name="tail" default="false" />
    <arg name="logging" default="false" />
    <arg name="parallel_index" default="0" />
    <arg name="twist_input" default="none" />

    <!-- Launch the node to publish the global body plan -->
    <group if="$(eval arg('global_planner') == 'fgmp')">
        <node name="global_body_planner" pkg="global_body_planner" type="global_body_planner_node" output="screen" />
        <param name="local_planner/use_twist_input" value="false" />
    </group>

    <!-- Use twist input to local plan and no global plan -->
    <group if="$(eval arg('global_planner') == 'twist')">
        <param name="local_planner/use_twist_input" value="true" /> <!-- Change value to false if don't want it to run twist command -->

        <group if="$(eval arg('twist_input') == 'keyboard')">
            <node name="teleop_twist_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" />
        </group>

        <group if="$(eval arg('twist_input') == 'joy')">
            <include file="$(find teleop_twist_joy)/launch/teleop.launch">
                <arg name="joy_config" value="ps3-holonomic" />
            </include>
        </group>
    </group>

    <node name="terrain_estimation" pkg="quad_utils" type="terrain_estimation_node" output="screen">
        <rosparam command="load" file="$(find quad_utils)/config/footstep_filter.yaml" />
    </node>

    <node name="local_planner" pkg="local_planner" type="local_planner_node" output="screen" />

    <!-- Launch the node to publish the tail plan -->
    <group if="$(arg tail)">
        <node name="tail_planner" pkg="tail_controller" type="tail_planner_node" output="screen" />
    </group>

    <group if="$(arg logging)">
        <include file="$(find quad_utils)/launch/logging.launch">
            <arg name="parallel_index" value="$(arg parallel_index)" />
        </include>
    </group>

</launch>